src/Utils/APITypeUtils.ts

```diff
@@ -0,0 +1,37 @@
+import { userContext } from "../UserContext";
+
+interface CollectionNameOptions {
+  isLowerCase?: boolean;
+  isPlural?: boolean;
+}
+
+export const getCollectionName = (options?: CollectionNameOptions): string => {
+  let collectionName: string;
+  switch (userContext.apiType) {
+    case "SQL":
+      collectionName = "Container";
+      break;
+    case "Mongo":
+      collectionName = "Collection";
+      break;
+    case "Cassandra":
+    case "Tables":
+      collectionName = "Table";
+      break;
+    case "Gremlin":
+      collectionName = "Graph";
+      break;
+    default:
+      throw new Error(`Unknown API type: ${userContext.apiType}`);
```

Comment:
If you handle the default like this:

```suggestion
    default: {
      const assertUnreachable = (type: never): never => {
        throw new Error(`Unknown API type: ${type}`);
      };
      assertUnreachable(userContext.apiType);
    }
```

then if a new possible value is added to `userContext.apiType`, then this code doesn't type-check anymore, which is great! See https://stackoverflow.com/a/39419171 for more.

Reply:
I wish there was a way to enforce we do this on all switch statements via linting

Reply:
That's very cool pattern! I'll start using this in the future.